File: app/models/player.rb
class Player 
  include Neo4j::ActiveNode


  property :name, type: String
  property :height_cm, type: Integer
  property :height_formatted, type: String
  property :weight_lb, type: Integer
  property :weight_kg, type: String
  property :position, type: String
  property :number, type: String, default: '-'
  property :birthplace, type: String
  property :birthdate, type: String
  property :updated_at
  property :created_at

  validates :name, :height_cm, :height_formatted,
    :height_cm, :birthdate, :birthplace, presence: true

  has_one :out, :real_team, model_class: NbaTeam
  has_many :out, :boxscores, model_class: BoxScore
  has_many :in, :contracts, model_class: Contract, origin: :player

end

File: blln/app/models/player.js
import DS from 'ember-data';
import PositionFormat from 'blln/mixins/positions';
import PlayerCharts from 'blln/mixins/player-charts';

export default DS.Model.extend(PositionFormat, PlayerCharts, {
  name: DS.attr('string'),
  heightFormatted: DS.attr('string'),
  heightCm: DS.attr('number'),
  weightLb: DS.attr('number'),
  weightKg: DS.attr('string'),
  birthplace: DS.attr('string'),
  birthdate: DS.attr('string'),
  position: DS.attr('string'),
  number: DS.attr('string'),
  stats: DS.attr(),
  nbaTeam: DS.belongsTo('nbaTeam', { async: true }),
  avgRebounds: function () {
    return (this.get('stats.defr') + this.get('stats.ofr')).toFixed(2);
  }.property('stats'),
  boxScores: DS.hasMany("box-score", { async: true }),
  contracts: DS.hasMany("contract", { async: true }),
  sharpId: function () {
    return "#" + this.get('id');
  }.property('sharpId'),
  positionLong: function () {
   return this.positionToWord(this.get('position'));
  }.property('position'),
  canPlay: function () {
    return this.positionsCanPlay(this.get('position'));
  }.property('position')
});